lib/Analysis/SchedulingAnalysis.cpp

```diff
@@ -76,19 +78,33 @@ void circt::analysis::CyclicSchedulingAnalysis::analyzeForOp(
   });
 
   // Insert conditional dependences into the problem.
-  forOp.getBody()->walk([&](AffineIfOp op) {
+  forOp.getBody()->walk([&](Operation *op) {
+    Block *thenBlock = nullptr;
+    Block *elseBlock = nullptr;
+    if (auto ifOp = dyn_cast<scf::IfOp>(op)) {
+      thenBlock = ifOp.thenBlock();
+      elseBlock = ifOp.elseBlock();
+    }
+    if (auto ifOp = dyn_cast<AffineIfOp>(op)) {
+      thenBlock = ifOp.getThenBlock();
+      if (ifOp.hasElse())
+        elseBlock = ifOp.getElseBlock();
+    }
+    if (!thenBlock)
```

Comment:
Nit: I think an if -- else if -- else would be a bit clearer here, as I wondered for a moment why `thenBlock` could be undefined.